src/kafka/connection/tcp_connection.hh

```diff
@@ -0,0 +1,68 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Copyright (C) 2019 ScyllaDB Ltd.
+ */
+
+#pragma once
+
+#include <seastar/core/future.hh>
+#include <seastar/net/net.hh>
+#include <string>
+
+namespace seastar {
+
+namespace kafka
+{
+
+class tcp_connection
+{
+
+public:
+
+    tcp_connection(const ipv4_addr& addr, connected_socket&& fd) noexcept
```

Comment:
it would be nicer to work on inet_address instead of ipv4_addr to allow support for ipv6 as well.